package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,331 @@
+            Dimension factor :math:`d` of the MSD.
+
+        """
+        keys = {'x':[0], 'y':[1], 'z':[2], 'xy':[0,1], 'xz':[0,2], 'yz':[1,2], 'xyz':[0,1,2]}
+
+        self.msd_type = self.msd_type.lower()
+
+        try:
+            self._dim = keys[self.msd_type]
+        except KeyError:
+            raise ValueError('invalid msd_type: {} specified, please specify one of xyz, xy, xz, yz, x, y, z'.format(self.msd_type))
+
+        self.dim_fac = len(self._dim)
+
+    def _single_frame(self):
+        r""" Constructs array of positions for MSD calculation.
+        
+        Parameters
+        ----------
+        self.u : :class:`Universe`
+            MDAnalysis Universe
+        
+        Returns
+        -------
+        self._position_array : :class:`np.ndarray`
+            Array of particle positions with respect to time shape = (n_frames, n_particles, 3)
+        """
+
+        self._position_array[self._frame_index,:,:] = self._atoms.positions
+            
+    def _conclude(self):
+        if self.fft == True:
+            self.timeseries = self._conclude_fft()
+        else:
+            self.timeseries = self._conclude_simple()
+
+    def _conclude_simple(self): 
+        r""" Calculates the MSD via the simple "windowed" algorithm.
+        
+        Parameters
+        ----------
+        self._position_array : :class:`np.ndarray`
+            Array of particle positions with respect to time shape (n_frames, n_particles, 3).
+
+        Returns
+        -------
+        self.timeseries : :class:`np.ndarray`
+            The MSD as a function of lag-time.
+
+        """
+        lagtimes = np.arange(1,self.n_frames)
+        self.msds_by_particle[0,:] = np.zeros(self._n_particles) # preset the zero lagtime so we don't have to iterate through
+        for lag in lagtimes:
+            disp = self._position_array[:-lag,:,self._dim] - self._position_array[lag:,:,self._dim]
+            sqdist = np.square(disp, dtype=np.float64).sum(axis=-1, dtype=np.float64) #accumulation in np.float64 required
+            self.msds_by_particle[lag,:] = np.mean(sqdist, axis=0, dtype=np.float64)
+        msd = self.msds_by_particle.mean(axis=1, dtype=np.float64)
+        return msd
+
+    def _conclude_fft(self): #with FFT, np.float64 bit prescision required.
+        r""" Calculates the MSD via the FCA fast correlation algorithm.
+        
+        Parameters
+        ----------
+        self._position_array : :class:`np.ndarray`
+            Array of particle positions with respect to time shape (n_frames, n_particles, 3).
+
+        Returns
+        -------
+        self.timeseries : :class:`np.ndarray`
+            The MSD as a function of lagtime.
+
+        """
+        reshape_positions = self._position_array[:,:,self._dim].astype(np.float64)
+        for n in range(self._n_particles):
+            self.msds_by_particle[:,n] = tidynamics.msd(reshape_positions[:,n,:])
+        msd = self.msds_by_particle.mean(axis=1, dtype=np.float64)
```

Comment:
pep8 - space after comma

Reply:
Nitpick -- [Don't repeat yourself](https://realpython.com/lessons/dry-principle/) by moving `self.timeseries` calculation back to `_conclude`.